backend/main/models/community_users.go

```diff
@@ -29,11 +37,25 @@ type CommunityUserPayload struct {
 	Composite_signatures *[]s.CompositeSignature `json:"compositeSignatures" validate:"required"`
 }
 
-func GetUsersForCommunity(db *s.Database, communityId, start, count int) ([]CommunityUser, int, error) {
-	var users = []CommunityUser{}
+func GetUsersForCommunity(db *s.Database, communityId, start, count int) ([]CommunityUserType, int, error) {
+	var users = []CommunityUserType{}
 	err := pgxscan.Select(db.Context, db.Conn, &users,
 		`
-		SELECT * FROM community_users WHERE community_id = $1
+		SELECT
```

Comment:
there might be a more efficient way to write this query but i'm not totally sure off the top of my head.

we can revisit later if its a problem. :)

Reply:
yeah I agree, we could have functions running each of the CASE statements to simply the query and maybe creating a view or a materialized view could help too.